terraso_backend/apps/project_management/graphql/projects.py

```diff
@@ -0,0 +1,351 @@
+            transfer_project = cls.get_or_throw(Project, "id", transfer_project_id)
+            if not user.has_perm(Project.get_perm("add"), transfer_project):
+                cls.not_allowed()
+            project_sites = project.site_set.all()
+            for site in project_sites:
+                site.project = transfer_project
+            Site.objects.bulk_update(project_sites, ["project"])
+        result = super().mutate_and_get_payload(root, info, **kwargs)
+        return result
+
+
+class ProjectArchiveMutation(BaseWriteMutation):
+    project = graphene.Field(ProjectNode, required=True)
+
+    model_class = Project
+
+    class Input:
+        id = graphene.ID(required=True)
+        archived = graphene.Boolean(required=True)
+
+    @classmethod
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, **kwargs):
+        user = info.context.user
+        project_id = kwargs["id"]
+        project = cls.get_or_throw(Project, "id", project_id)
+        if not user.has_perm(Project.get_perm("archive"), project):
+            cls.not_allowed()
+        project_sites = project.site_set.all()
+        for site in project_sites:
+            site.archived = kwargs["archived"]
+        Site.objects.bulk_update(project_sites, ["archived"])
+        result = super().mutate_and_get_payload(root, info, **kwargs)
+        return result
+
+
+class ProjectUpdateMutation(BaseWriteMutation):
+    project = graphene.Field(ProjectNode)
+
+    model_class = Project
+
+    class Input:
+        id = graphene.ID(required=True)
+        name = graphene.String()
+        privacy = graphene.Field(ProjectPrivacy)
+        description = graphene.String()
+
+    @classmethod
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, **kwargs):
+        logger = cls.get_logger()
+        user = info.context.user
+        project_id = kwargs["id"]
+        project = cls.get_or_throw(Project, "id", project_id)
+        if not user.has_perm(Project.get_perm("change"), project):
+            cls.not_allowed()
+        kwargs["privacy"] = kwargs["privacy"].value
+
+        metadata = {
+            "name": kwargs["name"],
+            "privacy": kwargs["privacy"],
+            "description": kwargs["description"] if "description" in kwargs else None,
+        }
+        logger.log(
+            user=user,
+            action=log_api.CHANGE,
+            resource=project,
+            client_time=datetime.now(),
+            metadata=metadata,
+        )
+
+        return super().mutate_and_get_payload(root, info, **kwargs)
+
+
+class ProjectAddUserMutation(BaseWriteMutation):
+    project = graphene.Field(ProjectNode, required=True)
+    membership = graphene.Field(MembershipNode, required=True)
+
+    model_class = Membership
+
+    class Input:
+        project_id = graphene.ID(required=True)
+        user_id = graphene.ID(required=True)
+        role = graphene.String(required=True)
```

Comment:
suggestion: could this be an enum instead of string field? same in `UpdateUserRoleMutation`

Reply:
Oh yeah, should definitely do that.